libsolidity/codegen/YulUtilFunctions.h

```diff
@@ -322,6 +323,37 @@ class YulUtilFunctions
 	static std::string revertReasonIfDebug(RevertStrings revertStrings, std::string const& _message = "");
 
 	std::string revertReasonIfDebug(std::string const& _message = "");
+
+	/// Generates the function-body and returns the function name for a function to decode the
+	/// return parameters.
```

Comment:
Please explain the signature of the function. Its behaviour is a bit special because it takes a memory pointer and moves the free-memory pointer beyond it (without checking that it is larger, for example).